src/porepy/viz/plot_grid.py

```diff
@@ -276,10 +276,19 @@ def plot_mdg(
         # Adjust rgb colors depending on the subdomain ordering
         kwargs["rgb"] = np.divide(kwargs.get("rgb", [1, 0, 0]), sd.id + 1)
         # Plot the subdomain and data
+
+        vector_value_array = sd_data.get(pp.STATE, {}).get(vector_value, None)
+        if vector_value_array is not None:
+            # The further algorithm requires the vector_value array of shape (3 x n). but
+            # We have the 1D data array. Thus, we fill the remaining dimensions with zeros.
+            vector_value_array = vector_value_array.reshape(sd.dim, -1, order="F")
```

Comment:
I have a feeling shape should be (mdg.dim_max(), -1) or perhaps rather (-1, sd.num_cells)

Reply:
(mdg.dim_max(), -1) - what if the domain is 3D, the subdomain and the data are 2D?
(-1, sd.num_cells) - the data can be defined at the edges.

Why don't you prefer my choice - (sd.dim, -1)?

Reply:
The vector is usually nd. E.g. traction on a fracture is a nd-vector, even though the fracture is 2d.
Does the vector plotting work for faces/edges?

Reply:
Alternatively, the dimension of the vector field can be made an optional input argument, with the default being (-1, sd.num_cells).

Reply:
Yes, if there's a (not too exotic) use case for that.

Reply:
> The vector is usually nd. E.g. traction on a fracture is a nd-vector, even though the fracture is 2d. Does the vector plotting work for faces/edges?

@IvarStefansson I see your point. I believe, (mdg.dim_max(), -1) must cover all possible cases. The vector plotting was intended to work with faces/edges, see: https://github.com/pmgbergen/porepy/blob/d4cc3adcbecf5401a5b2c2dcc4398e586cf5d9ff/src/porepy/viz/plot_grid.py#L395

@keileg probably, that will be superfluous. Taking mdg.dim_max() will always allow us to understand if the vector is placed in cell centres or edges etc. Or maybe there is some exotic use case I don't see now.

Reply:
I agree. If we ever need it we can always add a vector_dim-argument later.